pkg/controller/hyperconverged/factory.go

```diff
@@ -0,0 +1,149 @@
+package hyperconverged
+
+import (
+	sspv1 "github.com/MarSik/kubevirt-ssp-operator/pkg/apis/kubevirt/v1"
+	networkaddons "github.com/kubevirt/cluster-network-addons-operator/pkg/apis/networkaddonsoperator/v1alpha1"
+	networkaddonsnames "github.com/kubevirt/cluster-network-addons-operator/pkg/names"
+	hcov1alpha1 "github.com/kubevirt/hyperconverged-cluster-operator/pkg/apis/hco/v1alpha1"
+	kwebuis "github.com/kubevirt/web-ui-operator/pkg/apis/kubevirt/v1alpha1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	cdi "kubevirt.io/containerized-data-importer/pkg/apis/core/v1alpha1"
+	kubevirt "kubevirt.io/kubevirt/pkg/api/v1"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// The set of resources managed by the HCO
+func (r *ReconcileHyperConverged) getAllResources(cr *hcov1alpha1.HyperConverged, request reconcile.Request) []runtime.Object {
+	return []runtime.Object{
+		newKubeVirtConfigForCR(cr, request.Namespace),
```

Comment:
do we really need for these to be function calls when we can just list them here?

```
func () getAllResources(...) []runtime.Object {
	labels := map[string]string{
		"app": cr.Name,
	}
	return []runtime.Object{
		&corev1.ConfigMap{
			ObjectMeta: metav1.ObjectMeta{
				Name:      "kubevirt-config",
				Labels:    labels,
				Namespace: namespace,
			},
			Data: map[string]string{
				"feature-gates": "DataVolumes,SRIOV,LiveMigration,CPUManager,CPUNodeDiscovery",
			},
		},
	       ...
	}
}
```

Reply:
This contradicts my recommendation below :smile: 

I think having a collection of functions that individually reconcile their respective resources to the state they should be in is the correct way to go.